snakemake/shell.py

```diff
@@ -103,6 +110,16 @@ def __new__(
         env_modules = context.get("env_modules", None)
         shadow_dir = context.get("shadow_dir", None)
 
+        # environment variable list for linear algebra libraries taken from:
+        # https://stackoverflow.com/a/53224849/2352071
+        cls._process_prefix += (
```

Comment:
This will always extend process_prefix, making it longer and longer with each shell command. Instead, rather create a new variable `envvars` and add it to `cmd` below (line 123), after `cls._process_prefix`.